File: src/Deviser.Core/Deviser.Admin/Config/Filters/NumberFilter.cs
using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;

namespace Deviser.Admin.Config.Filters
{
    public class NumberFilter : Filter
    {
        [JsonConverter(typeof(StringEnumConverter))]
        public NumberOperator Operator { get; set; }
        public string Number { get; set; }
        public string FromNumber { get; set; }
        public string ToNumber { get; set; }
    }
}

File: src/Deviser.Core/Deviser.Admin.Web/ClientApp/src/app/common/domain-types/number-filter.ts
import { NumberOperator } from './number-operator';
import { Filter } from './filter';

export class NumberFilter implements Filter{
    fieldName: string;
    operator: NumberOperator;
    number: number;
    fromNumber: number;
    toNumber: number;
    constructor(fieldName: string) {
        this.fieldName = fieldName;
    }
}